File: src/test/java/jvm/testing/frameworks/GeometricShapeHelperTest.java
package jvm.testing.frameworks;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import java.util.stream.Stream;

import static org.assertj.core.api.Assertions.assertThat;
import static org.junit.jupiter.params.provider.Arguments.arguments;

class GeometricShapeHelperTest {
    GeometricShapeHelper geometricShapeHelper = new GeometricShapeHelper();

    @ParameterizedTest
    @MethodSource("shapes")
    @DisplayName("Calculate area by shape")
    void calculateArea(Shape shape, int expectedResult) {
        assertThat(geometricShapeHelper.calculateArea(shape)).isEqualTo(expectedResult);
    }

    static Stream<Arguments> shapes() {
        return Stream.of(
                arguments(new Square(5), 25),
                arguments(new Rectangle(3, 4), 12),
                arguments(new Rectangle(5, 1), 5)
        );
    }
}

File: src/test/kotlin/jvm/testing/frameworks/GeometricShapeHelperTest.kt
package jvm.testing.frameworks

import io.kotlintest.shouldBe
import io.kotlintest.specs.StringSpec
import io.kotlintest.tables.forAll
import io.kotlintest.tables.headers
import io.kotlintest.tables.row
import io.kotlintest.tables.table

class GeometricShapeHelperTest : StringSpec({
    val geometricShapeHelper = GeometricShapeHelper()

    "maximum of two numbers" {
        table(
                headers("shape", "expected result"),
                row(Square(5), 25),
                row(Rectangle(3, 4), 12),
                row(Rectangle(5, 1), 5)
        ).forAll { s, r ->
            geometricShapeHelper.calculateArea(s) shouldBe r
        }
    }
})